config/v3_2/config.go

```diff
@@ -56,6 +60,28 @@ func Parse(rawConfig []byte) (types.Config, report.Report, error) {
 	return config, rpt, nil
 }
 
-func isEmpty(userdata []byte) bool {
-	return len(userdata) == 0
+// ParseConfig parses the raw config of version 3.2.0 or lesser
+// into a 3.2 types.Config struct and generates a report of any
+// errors, warnings,info, and deprecations it encountered
+func ParseConfig(raw []byte) (types.Config, report.Report, error) {
+	version, rpt, err := util.GetConfigVersion(raw)
+	if err != nil {
+		return types.Config{}, rpt, err
+	}
+
+	switch version {
+	case types.MaxVersion:
+		return Parse(raw)
+	case types_3_1.MaxVersion, types_3_0.MaxVersion:
+		return v3_2_from_3_1(v3_1.ParseConfig(raw))
+	default:
+		return types.Config{}, report.Report{}, errors.ErrUnknownVersion
+	}
```

Comment:
The individual ParseConfig functions only need to care about two cases: 1) my version, and 2) not my version.  In case 2, configs &ge; 3.1 should call into the previous version, and config 3.0 should fail with `ErrUnknownVersion`.  That way, adding a new spec version doesn't require updating _all_ spec versions.